contrib/mesos/pkg/scheduler/podtask/pod_task.go

```diff
@@ -198,23 +165,59 @@ func (t *T) Has(f FlagType) (exists bool) {
 	return
 }
 
-func New(ctx api.Context, id string, pod *api.Pod) (*T, error) {
+func (t *T) Roles() []string {
+	var roles []string
+
+	if r, ok := t.Pod.ObjectMeta.Labels[annotation.RolesKey]; ok {
+		roles = strings.Split(r, ",")
+
+		for i, r := range roles {
+			roles[i] = strings.TrimSpace(r)
+		}
+
+		roles = filterRoles(roles, not(emptyRole), not(seenRole()))
+	}
+
+	if roles == nil {
```

Comment:
An empty roles label is different from a non-existing label. 

The first might be due to some tooling which explicitly does not allow any role. The scheduler should accept that.

The second means that the default roles are to be used.

In mathematical terms: we want continuity for the labels->roles function.